[one-cmds] Support group option (#10530)
Samsung/ONE: compiler/one-cmds/CMakeLists.txt

find_package(PythonInterp 3.8 QUIET)

> find_package(PythonLibs 3.8 QUIET)

Reviewer: We may have to use different python version for Ubuntu 18.04 vs Ubuntu 22.04
Is it possible to use just `3` instead of `3.8` ?

Author: I think just using `3` might happen to install the version lower than `3.8`. Current `PythonInterp` implicitly finds other candidate versions if there is no exact python version.

```bash
$ cat /usr/share/cmake-3.16/Modules/FindPythonInterp.cmake

set(_PYTHON1_VERSIONS 1.6 1.5)
set(_PYTHON2_VERSIONS 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0)
set(_PYTHON3_VERSIONS 3.9 3.8 3.7 3.6 3.5 3.4 3.3 3.2 3.1 3.0)

if(PythonInterp_FIND_VERSION)
    if(PythonInterp_FIND_VERSION_COUNT GREATER 1)
        set(_PYTHON_FIND_MAJ_MIN "${PythonInterp_FIND_VERSION_MAJOR}.${PythonInterp_FIND_VERSION_MINOR}")
        list(APPEND _Python_NAMES
             python${_PYTHON_FIND_MAJ_MIN}
             python${PythonInterp_FIND_VERSION_MAJOR})
        unset(_PYTHON_FIND_OTHER_VERSIONS)
        if(NOT PythonInterp_FIND_VERSION_EXACT)
            foreach(_PYTHON_V ${_PYTHON${PythonInterp_FIND_VERSION_MAJOR}_VERSIONS})
                if(NOT _PYTHON_V VERSION_LESS _PYTHON_FIND_MAJ_MIN)
                    list(APPEND _PYTHON_FIND_OTHER_VERSIONS ${_PYTHON_V})
                endif()
             endforeach()
        endif()
        unset(_PYTHON_FIND_MAJ_MIN)
    else()
        list(APPEND _Python_NAMES python${PythonInterp_FIND_VERSION_MAJOR})
        set(_PYTHON_FIND_OTHER_VERSIONS ${_PYTHON${PythonInterp_FIND_VERSION_MAJOR}_VERSIONS})
    endif()
else()
    set(_PYTHON_FIND_OTHER_VERSIONS ${_PYTHON3_VERSIONS} ${_PYTHON2_VERSIONS} ${_PYTHON1_VERSIONS})
endif()
find_program(PYTHON_EXECUTABLE NAMES ${_Python_NAMES})
```

_dalgona_ is already built with ubuntu 22.04 which has same cmake instructions.

Reviewer: Ah, I found that I've added this in other modules.
```
# NOTE find_package try to use at least python3.8 as follows depending on platform version
#   Ubuntu18.04; explictly installed python3.8 (default is python3.6)
#   Ubuntu20.04; default python3.8
#   Ubuntu22.04; default python3.10
#   refer https://github.com/Samsung/ONE/issues/9962
```
Can you please add this too ?

if(NOT ${PYTHONINTERP_FOUND})

>   message(STATUS "Build common-artifacts: FALSE (Python3 is missing)")

Reviewer 2: ```suggestion
  message(STATUS "Build one-lib: FALSE (Python3 is missing)")
```

  return()
endif()

if(${PYTHON_VERSION_MINOR} LESS 8)

>   message(STATUS "Build common-artifacts: FALSE (You need to install Python version higher than 3.8)")

Reviewer 2: ```suggestion
  message(STATUS "Build one-cmds: FALSE (You need to install Python version higher than 3.8)")
```

endforeach(ONE_PYTHON_FILE)

set(CONSTANT_EXPORTING_SCRIPT "${CMAKE_CURRENT_SOURCE_DIR}/${ONE_PYTHON_DIR}/export_constant.py")
set(O1_OPTION "O1")
set(O1_CFG_FILE "${O1_OPTION}.cfg")
set(O1_CFG_FILE_BIN "${CMAKE_CURRENT_BINARY_DIR}/${ONE_PYTHON_DIR}/${O1_CFG_FILE}")

add_custom_command(OUTPUT ${O1_CFG_FILE_BIN}

>   COMMAND ${PYTHON_EXECUTABLE} ${CONSTANT_EXPORTING_SCRIPT} --constant ${O1_OPTION}
>                                     --format cfg --output_path ${O1_CFG_FILE_BIN}

Reviewer: optional)
```suggestion
  COMMAND ${PYTHON_EXECUTABLE} ${CONSTANT_EXPORTING_SCRIPT}
          --constant ${O1_OPTION}
          --format cfg
          --output_path ${O1_CFG_FILE_BIN}
```

  DEPENDS ${CONSTANT_EXPORTING_SCRIPT}
  COMMENT "Generate ${O1_CFG_FILE}"
)

add_custom_target("O1_cfg_target" ALL DEPENDS ${O1_CFG_FILE_BIN})

install(FILES ${O1_CFG_FILE_BIN}

>         PERMISSIONS OWNER_WRITE OWNER_READ
>                          GROUP_READ
>                          WORLD_READ
>         DESTINATION optimization)

Reviewer: optional) 
```suggestion
        PERMISSIONS OWNER_WRITE OWNER_READ
                    GROUP_READ WORLD_READ
        DESTINATION optimization)
```

install(FILES ${O1_CFG_FILE_BIN}
        PERMISSIONS OWNER_WRITE OWNER_READ

>                     GROUP_READ WORLD_READ

Reviewer: ah.. sorry about this...
```suggestion
                    GROUP_READ
                    WORLD_READ
```
